File: Java/61. Rotate List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode rotateRight(ListNode head, int k) {
        if(k==0 || head==null){
            return head;
        }
        ListNode t=head;
        ListNode l=head;
        int count = 0;
        ListNode temp = head;
        while(temp!=null){
            count++;
            temp = temp.next;
        }
        for(int i = k%count-1; i >= 0; i--){
            while(t.next!=null){
                l=t;
                t=t.next;
            }
            l.next=null;
            t.next=head;
            head=t;
        }  
        return head;
    }
}

File: C++/61. Rotate List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* rotateRight(ListNode* head, int k) {
        if(k==0 || head==NULL){
            return head;
        }
        int count = 0;
        ListNode* t=head;
        ListNode* l=head;
        ListNode* temp = head;
        while(temp!=NULL){
            count++;
            temp = temp->next;
        }
        for(int i = k%count-1; i >= 0; i--){
            while(t->next!=NULL){
                l=t;
                t=t->next;
            }
            l->next=NULL;
            t->next=head;
            head=t;
        }  
        return head;
    }
};
